docs/developers-guide/substreams-sink-files.md

```diff
@@ -0,0 +1,275 @@
+---
+description: StreamingFast Substreams sink files
+---
+
+# `substreams-sink-files` introduction
+
+### Overview
+
+The `substreams-sink-files` tool provides the ability to pipe data extracted from a blockchain to various types of files-based persistence solutions. A few available options include [BigQuery](https://cloud.google.com/bigquery), [Clickhouse](https://clickhouse.com), custom scripts and other related data storage and processing tools accepting a file format. For example, you could extract all of the ERC20, ERC721, and ERC1155 transfers from the Ethereum blockchain and persist the data to a files-based store.
+
+Substreams modules are created and prepared for the sink tool. After the sink tool begins running, automated ingestion tasks can be setup to have BigQuery, or other files-based storage solution, ingest the data.
+
+As an example, you could use `substreams-sink-files` to sink data in `JSONL` format to a [Google Cloud Storage (GCS)](https://cloud.google.com/storage) bucket and configure a BigQuery Transfer job to run every 15 minutes. The scheduled job ingests the new files found in the GCS bucket where the data, extracted by the Substreams, was written.
+
+### Accompanying code example
+
+[The accompanying Substreams module](#) associated with this documentation is responsible for extracting a handful of data fields from the Block object injected into the Rust-based map module. The sink tool processes the extracted blockchain data line-by-line and outputs the data to the files-based persistence mechanism you've chosen.
```

Comment:
I've uploaded the code to https://github.com/streamingfast/substreams-sink-files/tree/master/docs/tutorial. I've made some quick changes:

- The Substreams manifest is always named `substreams.yaml`
- We are going to use `xxx_out` for module which are output module for `sink` consumption, I've just renamed it to `jsonl_out` (please update tutorial accordingly)
- The Protobuf output module is now `sf.substreams.sink.files.v1`, tutorial has been updated.

Reply:
Note: This breaks the StreamingFast prescribed naming conventions for map modules. It's mentioned elsewhere in the documentation.

Note: Readers will have to clone the sink files repo with the sink tool and the code example inside of the tool repo. The current documentation directs them to a standalone repo to clone just the code example. I'll rework the current approach so they download the entire repo and then access the code example rather than clone it individually.

Reply:
> Note: This breaks the StreamingFast prescribed naming conventions for map modules. It's mentioned elsewhere in the documentation.

Yeah that is true hence why I'm noting the change here.

> Note: Readers will have to clone the sink files repo with the sink tool and the code example inside of the tool repo. The current documentation directs them to a standalone repo to clone just the code example. I'll rework the current approach so they download the entire repo and then access the code example rather than clone it individually.

That is true but I think it's ok. I have added it there in the hope that it does not rot and become stalled.

Reply:
Ok, no problem. Thought it might be a good idea to document those thoughts :)